src/timearray.jl

```diff
@@ -199,16 +199,20 @@ function getindex{T,N,D}(ta::TimeArray{T,N,D}, d::D)
     idxs = searchsorted(ta.timestamp, d)
     length(idxs) == 1 ? ta[idxs[1]] : nothing
 end
- 
+
 # multiple dates
 function getindex{T,N,D}(ta::TimeArray{T,N,D}, dates::Vector{D})
     dates = sort(dates)
     idxs, _ = overlaps(ta.timestamp, dates)
     ta[idxs]
 end #getindex
 
+# StepRange{Date,...}
 getindex{T,N,D}(ta::TimeArray{T,N,D}, r::StepRange{D}) = ta[collect(r)]
 
+# StepRange{Int,Int}
+getindex{T,N}(ta::TimeArray{T,N}, r::StepRange{Int,Int}) = ta[collect(r)]
```

Comment:
However, a better solution may be to refactor the existing methods to widen the signatures to `AbstractVector` for `Integer` subtypes.